unittest/CMakeLists.txt

```
@@ -1,5 +1,11 @@
 include(GTest)
 
+if(BUILD_MPI)
+  function(add_mpi_test name binary nproc)
+    add_test(NAME ${name} COMMAND ${MPIEXEC_EXECUTABLE} -np ${nproc} ${binary})
```

```suggestion
    add_test(NAME ${name} COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${nproc} ${MPIEXEC_PREFLAGS} ${binary} ${MPIEXEC_POSTFLAGS})
```

Newer OpenMPI version disable able oversubscription by default, hence allowing `-DMPIEXEC_PREFLAGS='--oversubscribe'` is sometimes useful. `MPIEXEC_NUMPROC_FLAG` is just a more generic way to write `-np`.

---

@junghans made some adjustments to make `add_mpi_test` a bit more like `add_test`, adding options like `NAME`, `COMMAND`, `WORKING_DIRECTORY` and `NUM_PROCS`. I also incorporated the `MPIEXEC` variables you mentioned.